x-pack/plugins/observability/server/services/slo/tasks/summary_cleanup_task.ts

```diff
@@ -0,0 +1,212 @@
+      let sloSummaryIdsToDelete: string[] = [];
+
+      for await (const response of finder.find()) {
+        const soItems = response.saved_objects.map(
+          (so) => `${so.attributes.id}${SEPARATOR}${so.attributes.revision}`
+        );
+
+        const { sloSummaryIds, searchAfter } = await this.fetchSloSummariesIds(searchAfterKey);
+        searchAfterKey = searchAfter;
+
+        const { missingSummaryIds, missingSOIds } = findMissingItems(
+          soItems.concat(soIdsToCheck),
+          sloSummaryIds.concat(sloSummaryIdsToDelete)
+        );
+
+        soIdsToCheck = missingSOIds;
+        sloSummaryIdsToDelete = missingSummaryIds;
+        if (searchAfterKey === undefined) {
+          break;
+        }
+      }
+
+      await finder.close();
+
+      if (sloSummaryIdsToDelete.length > 0) {
+        this.logger.info(
+          `[SLO] Deleting ${sloSummaryIdsToDelete.length} SLO Summaries from the summary index`
+        );
+        this.logger.info(JSON.stringify(sloSummaryIdsToDelete));
+
+        await this.esClient.deleteByQuery({
+          index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+          query: {
+            bool: {
+              should: getDeleteQueryFilter(sloSummaryIdsToDelete.sort()),
+            },
+          },
+        });
+      }
+    }
+
+    return { state: { lastRunAt: runAt } };
+  };
+
+  fetchSloSummariesIds = async (searchAfter?: SortResults) => {
+    if (this.esClient) {
+      const result = await this.esClient.search({
+        size: 2000,
+        index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+        fields: ['slo.id', 'slo.revision'],
+        collapse: {
+          field: 'slo.id',
+        },
+        _source: false,
+        sort: [
+          {
+            'slo.id': {
+              order: 'desc',
+            },
+            'slo.revision': {
+              order: 'desc',
+            },
+            'slo.instanceId': {
+              order: 'desc',
+            },
+          },
+        ],
+        search_after: searchAfter,
+      });
+
+      const newSearchAfter = result.hits.hits[result.hits.hits.length - 1].sort;
+      return {
+        searchAfter: newSearchAfter,
+        sloSummaryIds: result.hits.hits.map(
+          ({ fields }) =>
+            `${fields?.['slo.id'][0]}${SEPARATOR}${fields?.['slo.revision'][0]}` as string
+        ),
+      };
+    } else {
+      return {
+        searchAfter: undefined,
+        sloSummaryIds: [] as string[],
+      };
+    }
+  };
+
+  private get taskId() {
+    return `${TASK_TYPE}:1.0.0`;
+  }
+
+  public async start(
+    taskManager: TaskManagerStartContract,
+    soClient: SavedObjectsClientContract,
+    esClient: ElasticsearchClient
+  ) {
+    this.taskManager = taskManager;
+    this.soClient = soClient;
+    this.esClient = esClient;
+
+    if (!taskManager) {
+      this.logger.info('[SLO] Missing required service during startup, skipping task.');
+    }
+
+    this.taskManager.removeIfExists(this.taskId);
```

Comment:
What does this thing do? I wonder if I should do something similar for the enrich policy task

Reply:
this is only needed when testing interval. if you change schedule it won't apply. Alternatively without using this you can return interval in state when task runs and that also updates the interval.